components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/validators/TokenValidationHandlerTest.java

```diff
@@ -0,0 +1,374 @@
+ */
+
+package org.wso2.carbon.identity.oauth2.validators;
+
+import org.mockito.Mock;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.testng.PowerMockTestCase;
+import org.testng.IObjectFactory;
+import org.testng.annotations.ObjectFactory;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.oauth.config.OAuthServerConfiguration;
+import org.wso2.carbon.identity.oauth2.dto.OAuth2TokenValidationRequestDTO;
+import org.wso2.carbon.identity.oauth2.dto.OAuth2TokenValidationResponseDTO;
+import org.wso2.carbon.identity.oauth2.model.AccessTokenDO;
+import org.wso2.carbon.identity.oauth2.util.OAuth2Util;
+import org.wso2.carbon.user.core.util.UserCoreUtil;
+
+import java.sql.Timestamp;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.powermock.api.mockito.PowerMockito.doThrow;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.spy;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.powermock.api.mockito.PowerMockito.whenNew;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNotNull;
+
+@PrepareForTest({OAuthServerConfiguration.class, TokenValidationHandler.class, UserCoreUtil.class,
+        AccessTokenDO.class, OAuth2Util.class})
+public class TokenValidationHandlerTest  extends PowerMockTestCase {
+
+    AccessTokenDO accessTokenDO;
+
+    @Mock
+    private TokenValidationHandler mockTokenValidationHandler;
+
+    @Mock
+    private OAuth2TokenValidationRequestDTO mockOAuth2TokenValidationRequestDTO;
+
+    @Mock
+    private OAuth2TokenValidationRequestDTO.OAuth2AccessToken mockAuth2AccessToken;
+
+    @Mock
+    private OAuth2TokenValidator mockOAuth2TokenValidator;
+
+    @Mock
+    private OAuthServerConfiguration mockOAuthServerConfiguration;
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
+        return new org.powermock.modules.testng.PowerMockObjectFactory();
+    }
+
+    @Test
+    public void testGetInstance() throws Exception {
+        whenNew(TokenValidationHandler.class).withNoArguments().thenReturn(mockTokenValidationHandler);
+        assertNotNull(TokenValidationHandler.getInstance());
+    }
+
+    @Test
+    public void testAddTokenValidator() throws Exception {
+        mockOAuthServerConfiguration();
+        TokenValidationHandler.getInstance().addTokenValidator("type", mockOAuth2TokenValidator);
+    }
+
+    @Test
+    public void testValidate() throws Exception {
+        mockOAuthServerConfiguration();
+        TokenValidationHandler.getInstance().validate(new OAuth2TokenValidationRequestDTO());
+        assertNotNull(TokenValidationHandler.getInstance().validate(new OAuth2TokenValidationRequestDTO()
+        ), "Error while calling validate()");
```

Comment:
In addition to assertNotNull, I think we should also add two more assertions for true/false cases.